DEVELOPER.md

```diff
@@ -0,0 +1,359 @@
+Check out the [example resource agent](agent/resource-agent/examples/example_resource_agent/main.rs) to see how to create a resource agent.
+
+### [Bottlerocket](bottlerocket)
+
+The bottlerocket directory of the repo is broken into 3 sections: agents, testsys, and types.
+
+[agents](bottlerocket/agents) contains the implementations of all bottlerocket agents.
+
+The types used by the agents can be found in [types](bottlerocket/types).
+
+[testsys](bottlerocket/testsys) contains a cli implementation that is closely related to bottlerocket agents but will be removed in the near future.
+
+### [Cli](cli)
+
+Cli is and example command line interface that uses the [TestManager](model/src/test_manager/manager.rs) to interact with a testsys cluster.
+
+### [Controller](controller)
+
+The controller orchestrates the the interaction between test and resource. For more info see the [design](design/DESIGN.md) document. 
+
+### [Model](model)
+
+The model library contains the infrastructure that testsys is built upon.
+This includes the crds for [tests](model/src/test.rs) and [resources](model/src/resource.rs) as well as [everything](model/system) needed to be installed to the testsys cluster.
+
+All interactions with the testsys cluster should be done using the model library through `TestManager`, `TestClient`, or `ResourceClient`.
+
+The model provides [TestManager](model/src/test_manager/manager.rs) which can be used as an entry point for user interactions with testsys.
+To see how to use the TestManager look at the [example cli](cli).
+
+Model also contains a [TestClient](model/src/clients/test_client) for simple interactions with `Test`s and a [ResourceClient](model/src/clients/resource_client) for `Resource`s.
+
+### [Selftest](selftest)
+
+Selftest is a library used to validate incoming pull request by testing the overall testsys framework.
+
+## Getting Started
+
+Testsys is currently tested on `x86_64` machines, these steps may not work as expected on other operating systems.
+
+### Setting up testsys
+
+#### Create a testsys directory
+Create a directory that you wish use as a working directory and navigate to it. 
+The example provided creates a directory in the users home, but a directory may be created anywhere.
+
+```bash
+cd ~
+mkdir testsys && cd testsys
+pwd 
+```
+
+#### Clone the testsys repo
+While in the testsys directory, clone the `bottlerocket-test-system` repo from github.
+
+```bash
+git clone https://github.com/bottlerocket-os/bottlerocket-test-system.git
+cd bottlerocket-test-system
+export TESTSYS_DIR=$(pwd)
+```
+
+The testsys directory is now set up.
+
+### Installing tools
+
+#### Install Kind
+Instead of creating a long-standing eks cluster for testsys development, kind allows the quick creation and destruction of k8s clusters which helps improve the testing experience.
```

Comment:
It may be useful to indicate that `kind` is for _local_ clusters that exist within the developers local Docker